daemon/redis.c

```diff
@@ -1363,6 +1677,160 @@ static int redis_link_maps(struct redis *r, struct call *c, struct redis_list *m
 	return 0;
 }
 
+static void json_restore_call(struct redis *r, struct callmaster *m, redisReply *id, enum call_type type) {
+	redisReply* rr_jsonStr;
+	struct redis_hash call;
+	struct redis_list tags, sfds, streams, medias, maps;
+	struct call *c = NULL;
+	str s ;
+	const char *err = 0;
+	int i;
+#if !(GLIB_CHECK_VERSION(2,36,0))
+	g_type_init();
+#endif
+	JsonReader *root_reader =0;
+	JsonParser *parser =0;
+	char orig_json_callid[512]; memset(orig_json_callid,0,512);
+	memcpy(orig_json_callid,id->str,strlen(id->str));
```

Comment:
This seems unnecessary if str_cut is removed since the string then isn't modified. I would suggest renaming `id` to something like `full_json_id`, then declare your own `id`
```
str id;
str_init_redis(&id, full_json_id);
str_shift(&id, strlen("json-"));
```
Which actually duplicates the `s` variable below, which then becomes redundant.

This would require updating the function signature of all called functions to receive a `str` instead of a `redisReply`.  In fact, looking at the functions, they all seem to work on the just the call-id anyway, and do receive the `call` object, so that parameter could be removed completely and the functions could just use `call->callid`. If I'm not mistaken.